src/java.base/share/classes/jdk/internal/constant/ReferenceClassDescImpl.java

```diff
@@ -46,12 +50,23 @@ final class ReferenceClassDescImpl implements ClassDesc {
      * field descriptor string, or does not describe a class or interface type
      * @jvms 4.3.2 Field Descriptors
      */
-    ReferenceClassDescImpl(String descriptor) {
+    public static ReferenceClassDescImpl of(String descriptor) {
         int dLen = descriptor.length();
         int len = ConstantUtils.skipOverFieldSignature(descriptor, 0, dLen, false);
         if (len <= 1 || len != dLen)
             throw new IllegalArgumentException(String.format("not a valid reference type descriptor: %s", descriptor));
-        this.descriptor = descriptor;
+        return new ReferenceClassDescImpl(descriptor);
+    }
+
+    /**
+     * Creates a {@linkplain ClassDesc} from a pre-validated descriptor string
+     * for a class or interface type or an array type.
+     *
+     * @param descriptor a field descriptor string for a class or interface type
+     * @jvms 4.3.2 Field Descriptors
+     */
+    public static ReferenceClassDescImpl ofTrusted(String descriptor) {
```

Comment:
If you named `ofTrusted` to be in parity with the MTDImpl factory, they are different as MTD one means array is trusted; this one means descriptor is already valid.

Reply:
`ofValidated`?

Reply:
Sounds good to me, as long as the 2 factories take any different name combination so we don't think they both bypass everything